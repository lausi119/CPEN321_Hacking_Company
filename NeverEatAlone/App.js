import React from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { AppLoading, Asset, Font, Icon } from 'expo';
import AppNavigator from './navigation/AppNavigator';
import Geolocation from 'react-native-geolocation-service';


export default class App extends React.Component {
  state = {
    isLoadingComplete: false,
    hasLocationPermission: false,
    refreshInterval: null,
  };
  getPosition(){
    Geolocation.getCurrentPosition(
      (position) => {
        global.location = {
          "lat": position.coords.latitude,
          "long": position.coords.longitude,
        }
      },
      (error) => {
        console.log(error.message);
      },
      { enableHighAccuracy: true, 
        timeout: 15000, maximumAge: 10000}
    );
  }
  uploadPosition(){

  }
  refreshAll(){
    this.getPosition();
    this.uploadPosition();
  }
  startRefresh() {
    this.refreshAll();
    this.state.refreshInterval = 
      setInterval(this.refreshAll.bind(this),60*1000);
  };
  stopRefresh() {
    clearInterval(this.state.refreshInterval);
  }
  componentDidMount(){
    global.startRefresh = this.startRefresh.bind(this);
    global.stopRefresh = this.stopRefresh.bind(this);
  }
  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <View style={styles.container}>
          {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
          <AppNavigator />
        </View>
      );
    }
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./assets/images/robot-dev.png'),
        require('./assets/images/robot-prod.png'),
      ]),
      Font.loadAsync({
        // This is the font that we are using for our tab bar
        ...Icon.Ionicons.font,
        // We include SpaceMono because we use it in HomeScreen.js. Feel free
        // to remove this if you are not using it in your app
        'space-mono': require('./assets/fonts/SpaceMono-Regular.ttf'),
      }),
    ]);
  };

  _handleLoadingError = error => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
    global.location = "null";
  };
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
});
