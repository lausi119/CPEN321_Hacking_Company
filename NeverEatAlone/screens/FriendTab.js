import React from "react";
import { View,ScrollView, StyleSheet, Text, ListView,
  TouchableOpacity,
  ActivityIndicator} from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import { ExpoLinksView } from "@expo/samples";
import { Platform } from "react-native";
import ReactObserver from 'react-event-observer';

const styles = StyleSheet.create({
  loading:{
    alignItems: "center",
    position: "absolute",
    top: 50,
    right: 0,
    left: 0,
    bottom: 0,
  },
  text: {
    fontSize: 26,
    color: "#4f603c",
    paddingLeft: 10,
  },
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: "#fff",
  },
  headline:{
    flex: 1,
    textAlign: "center",
    fontSize: 20,
    bottom: 0,
    padding: 10,
    fontWeight: "bold"
  },
  header: {
    flexDirection: "row",
    borderBottomWidth: 1,
    borderColor: "#ddd",
    height: 50,
    backgroundColor: "#fff"
  },
  green: {
    backgroundColor: "green",
  },
  red: {
    backgroundColor: "red",
  },
  h2: {
    flex: 1,
    paddingTop: 15,
    paddingLeft: 15,
    marginBottom: 7,
    fontSize: 26,
    backgroundColor: "#fff",
  },
  circle: {
    marginTop: 17,
    marginRight: 25,
    width: 15,
    height: 15,
    borderRadius: 14,
  },
  friend: {
    borderRadius: 14,
    borderColor: "#000",
    borderWidth: 0.5,
    paddingLeft: 8,
    marginLeft: 15,
    paddingTop: 5,
    paddingBottom: 5,
  },
  menuOption: {
    flexDirection: "row",
    borderRadius: 14,
    borderColor: "#000",
    borderWidth: 0.5,
    paddingLeft: 8,
    marginLeft: 15,
    paddingTop: 5,
    paddingBottom: 5,
    marginTop: 5,
  },
  icon: {
    padding: 4,
    fontSize: 36,
    color: "#4f603c"
  },
});

export default class FriendTab extends React.Component {

  constructor(props){
    super(props);
    this.state = {
      /*
      1: FriendList
      2: Venue type options
      3: Venue select
      */
      screen: 1, 
      friendsOnline:[],
      friendsBusy:[],
      finishedLoading: true,
      title: "Friends",
      selectedFriend: {
      },
    };
    if(props.friendsOnline){
      this.state.friendsOnline=props.friendsOnline;
    }
    if(props.friendsBusy){
      this.state.friendsBusy=props.friendsBusy;
    }
    var updateFriends = function(data){
      this.setState((previousState) => {
        var newState = previousState;
        newState.friendsOnline = data;
        return {newState};
      });
    }.bind(this);
    this.listener = global.observer.subscribe('updateFriends',updateFriends);
  }
  locationDifference(loc1,loc2){
    return Math.sqrt(Math.abs(loc1.lat-loc2.lat),
      Math.abs(loc1.long-loc2.long));
  }
  static navigationOptions = {
    header: null,
  };
  chooseVenueType(type){
    //yelp.api.call??
    this.setState((previousState) => {
      var newState = previousState;
      newState.screen = 3;
      newState.finishedLoading = false;
      return {newState};
    });
  }
  chooseFriend(item, online){
    item["online"] = online;
    this.setState((previousState) => {
      var newState = previousState;
      newState.screen = 2;
      newState.selectedFriend = item;
      return {newState};
    });
  }
  resetScreen(){
    this.setState((previousState) => {
      var newState = previousState;
      newState.screen = 1;
      newState.selectedFriend = {};
      return {newState};
    });
  }

  waiting(){
    if(this.state.finishedLoading){
      return <ActivityIndicator
        size="large"
        color="#000"
      />;
    }
  }

  componentWillUnmount(){
    this.listener.unsubscribe();
  }

  render() {
    if(!global.finishedLoading){
      return <View style={styles.loading}>
        <ActivityIndicator
          color="#000"
          size="large"
      />
      </View>;
    }
    //CHOOSE TYPE OF VENUE/MESSAGE
    else if(this.state.screen === 2){
      return (<View style={styles.container}>
      <View style={styles.header}>
      <Icon 
        onPress={this.resetScreen.bind(this)}
        style={styles.icon}
        name = {Platform.OS === "ios"
          ? "ios-arrow-dropleft"
          : "md-arrow-dropleft"}
      />
        <Text style={styles.headline}
        >{this.state.selectedFriend.firstName}</Text>
        <View style={[
          this.state.selectedFriend.online ?
          styles.green: styles.red,
          styles.circle]}/>
      </View>
        <View>
        <TouchableOpacity style={styles.menuOption}
        onPress={() => this.chooseVenueType("Restaurant")}>
        <Icon name={Platform.OS === "ios"
        ? "ios-restaurant"
        : "md-restaurant"}
        size={40}/>
        <Text style={styles.text}>
          Restaurants
        </Text></TouchableOpacity>
        <TouchableOpacity style={styles.menuOption}
        onPress={() => this.chooseVenueType("Cafe")}>
        <Icon name={Platform.OS === "ios"
        ? "ios-cafe"
        : "md-cafe"}
        size={40}/>
        <Text style={styles.text}>
          Cafes
        </Text></TouchableOpacity>
        <TouchableOpacity style={styles.menuOption}
        onPress={() => this.chooseVenueType("Bar")}
        ><Icon name={Platform.OS === "ios"
        ? "ios-wine"
        : "md-wine"}
        size={40}/>
        <Text style={styles.text}>
          Bars
        </Text></TouchableOpacity>
        </View>
      </View>);
    }
    //CHOOSE VENUE 
    else if(this.state.screen === 3){
      return (
        <View style={styles.container}>
        <View style={styles.header}>
          <Icon 
            onPress={() => this.chooseFriend(this.state.selectedFriend,this.state.selectedFriend.online)}
            style={styles.icon}
            name = {Platform.OS === "ios"
              ? "ios-arrow-dropleft"
              : "md-arrow-dropleft"}
          />
        <Text style={styles.headline}
        >{this.state.selectedFriend.firstName}</Text>
        <View style={[
          this.state.selectedFriend.online ?
          styles.green: styles.red,
          styles.circle]}/>
      </View>
        <View style={styles.container}>
        {this.waiting()}
        </View>
        </View>
      );
    }
    //FRIEND LIST
    else {
      return (
      <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headline}
        >Friends</Text>
      </View>
      <ScrollView style={styles.container}>
      <Text style={styles.h2}>{this.state.selectedFriend.id}</Text>
      
      <Text style={styles.h2}>Available Nearby</Text>
      <View>
          {
            this.state.friendsOnline.map((item,index) => (
              <TouchableOpacity
              key={item.id}
              style={styles.friend}
              onPress={() => this.chooseFriend(item,true)}>
              <Text style={styles.text}>
                  {item.name}
              </Text>
              </TouchableOpacity>
          ))
          }
      </View>
      <Text style={styles.h2}>Busy Nearby</Text>
      <View>
          {
            this.state.friendsBusy.map((item,index) => (
              <TouchableOpacity
              key={item.id}
              style={styles.friend}
              onPress={() => this.chooseFriend(item,false)}>
              <Text style={styles.text}>
                  {item.name}
              </Text>
              </TouchableOpacity>
          ))
          }
      </View>
      </ScrollView>
      </View>);
    }
  }
}