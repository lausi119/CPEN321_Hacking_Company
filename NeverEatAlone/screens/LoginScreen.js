import React from "react";
import {
    Image,
    ScrollView,
    StyleSheet,
    Text,
    TouchableOpacity,
    View, Platform
  } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    justifyContent: "center",
    alignItems: "center",
  },
  loginButton: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "column",
  }
});

export default class LoginScreen extends React.Component {
  constructor(props){
    super(props);
    this.loggingIn = false;
  }
  static navigationOptions = {
    header: null,
  };

  async login(){
    global.userInfo = {
      'accessToken': '',
      'friends': [],
    };
    // Uncomment this line only when testing app without login. Comment everything below out if you do
    //this.props.navigation.navigate("App");
    const { type, token } = await 
    Expo.Facebook.logInWithReadPermissionsAsync("305115093422180", {
      permissions: ["public_profile","email","user_friends"],
    });
    if (type === "success") {
      global.userInfo = {
        'accessToken': token,
      };
      global.loggedIn = true;
      global.startRefresh();
      this.loggingIn = true;
      this.props.navigation.navigate("App");
      this.loggingIn = false;
    }
  } 

  render() {
    if(this.loggingIn){
      return <View style={styles.loading}>
        <ActivityIndicator
          color="#000"
          size="large"
      />
      </View>;
    }
    else {
      return (
        <View style={styles.container}>
        <View style={styles.loginButton}>
           <Text>LOGIN</Text>
          <TouchableOpacity
           id="login-button"
           onPress={() => this.login()}>
          <Icon
           name={Platform.OS === "ios"
            ? "ios-log-in"
            : "md-log-in"}
            size={50}
            />
          </TouchableOpacity>
          </View>
        </View>
      );
    }
  }
}