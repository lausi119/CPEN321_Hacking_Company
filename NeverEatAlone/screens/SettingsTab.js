import React from "react";
import{
  Image,
  ScrollView,StyleSheet,
  Text,TouchableOpacity,
  View, Button,
  Platform,
} from "react-native";
import Icon from "react-native-vector-icons/Ionicons";

const styles = StyleSheet.create({
  icon: {
    marginLeft: 10,
    marginTop: 5,
  },
  text: {
    paddingLeft: 10,
    paddingTop: 10,
  },
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: "#fff",
  },
  header: {
    flexDirection: "row",
    borderBottomWidth: 1,
    borderColor: "#ddd",
    height: 50,
    backgroundColor: "#fff"
  },
  headline:{
    flex: 1,
    textAlign: "center",
    fontSize: 20,
    bottom: 0,
    padding: 10,
    fontWeight: "bold"
  },
  listItemA:{
    flex: 1,
    flexDirection: "row",
    borderTopWidth: 1,
    borderColor: "black",
    backgroundColor: "#cccccc",
    height: 35,
    paddingLeft: 10
  },
  listItemB:{
    flex: 1,
    flexDirection: "row",
    borderTopWidth: 1,
    borderColor: "black",
    backgroundColor: "#eeeeee",
    height: 35,
    paddingLeft: 10
  }
});

export default class SettingsTab extends React.Component {
  navigation = {}
  constructor(props){
    super(props);
    this.navigation = props.navigation;
  }
  static navigationOptions = {
    header: null,
  };

  logout(){
    global.userInfo = null,
    global.stopRefresh();
    global.loggedIn = false;
    this.props.navigation.navigate("Auth");
  }

  render() {
    /* Go ahead and delete ExpoConfigView and replace it with your
     * content, we just wanted to give you a quick view of your config */
    return <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headline}
        >Settings</Text>
      </View>
      <View style={styles.container}>
        <ScrollView>
          <View style={styles.listItemA}>
          <Text>{global.id}</Text>
          </View>
          <TouchableOpacity style={styles.listItemB}
            onPress={this.logout.bind(this)}>
            <Text style={styles.text}
            >LOGOUT</Text>
            <Icon style={styles.icon}
            name={Platform.OS === "ios"
            ? "ios-log-out"
            : "md-log-out"}
              size={25}
            />
          </TouchableOpacity>
        </ScrollView>
      </View>
    </View>;
  }
}