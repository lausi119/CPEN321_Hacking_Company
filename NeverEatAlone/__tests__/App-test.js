import 'react-native';
import React from 'react';
import App from '../App';
import LoginScreen from '../screens/LoginScreen';
import FriendTab from '../screens/FriendTab';
import SettingsTab from '../screens/SettingsTab';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import {shallow} from 'enzyme';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure( {adapter: new Adapter() } );

describe('App snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    //NavigationTestUtils.resetInternalState();
  });

  it('renders the login screen', async () => {
    const component = shallow(<LoginScreen/>);
    expect(component).toMatchSnapshot();
  });
  
  it('logs in with test user', async () => {
    const component = shallow(<LoginScreen/>);
	var btn = component.find('#login-button');
	btn.simulate('click');
    const landing = shallow(<FriendTab/>);
    expect(component).toMatchSnapshot();
  });
});
