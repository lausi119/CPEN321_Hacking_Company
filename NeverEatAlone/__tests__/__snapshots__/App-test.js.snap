// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot logs in with test user 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          source={1}
          style={
            Object {
              "marginBottom": 100,
            }
          }
        />,
        <TouchableOpacity
          activeOpacity={0.2}
          id="login-button"
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3b5999",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          &gt;
          <Image
            source={1}
            style={
              Object {
                "margin": 8,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "margin": 8,
              }
            }
          >
            LOGIN THROUGH FACEBOOK
          </Text>
        </TouchableOpacity>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": 1,
          "style": Object {
            "marginBottom": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            ">",
            <Image
              source={1}
              style={
                Object {
                  "margin": 8,
                }
              }
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "margin": 8,
                }
              }
            >
              LOGIN THROUGH FACEBOOK
            </Text>,
          ],
          "id": "login-button",
          "onPress": [Function],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#3b5999",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          ">",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
              "style": Object {
                "margin": 8,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "LOGIN THROUGH FACEBOOK",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
                "margin": 8,
              },
            },
            "ref": null,
            "rendered": "LOGIN THROUGH FACEBOOK",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={1}
            style={
              Object {
                "marginBottom": 100,
              }
            }
          />,
          <TouchableOpacity
            activeOpacity={0.2}
            id="login-button"
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3b5999",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            &gt;
            <Image
              source={1}
              style={
                Object {
                  "margin": 8,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "margin": 8,
                }
              }
            >
              LOGIN THROUGH FACEBOOK
            </Text>
          </TouchableOpacity>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "marginBottom": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              ">",
              <Image
                source={1}
                style={
                  Object {
                    "margin": 8,
                  }
                }
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "margin": 8,
                  }
                }
              >
                LOGIN THROUGH FACEBOOK
              </Text>,
            ],
            "id": "login-button",
            "onPress": [Function],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "#3b5999",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            ">",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
                "style": Object {
                  "margin": 8,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "LOGIN THROUGH FACEBOOK",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "margin": 8,
                },
              },
              "ref": null,
              "rendered": "LOGIN THROUGH FACEBOOK",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App snapshot renders the login screen 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          source={1}
          style={
            Object {
              "marginBottom": 100,
            }
          }
        />,
        <TouchableOpacity
          activeOpacity={0.2}
          id="login-button"
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3b5999",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          &gt;
          <Image
            source={1}
            style={
              Object {
                "margin": 8,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "margin": 8,
              }
            }
          >
            LOGIN THROUGH FACEBOOK
          </Text>
        </TouchableOpacity>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": 1,
          "style": Object {
            "marginBottom": 100,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            ">",
            <Image
              source={1}
              style={
                Object {
                  "margin": 8,
                }
              }
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "margin": 8,
                }
              }
            >
              LOGIN THROUGH FACEBOOK
            </Text>,
          ],
          "id": "login-button",
          "onPress": [Function],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#3b5999",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          ">",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
              "style": Object {
                "margin": 8,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "LOGIN THROUGH FACEBOOK",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
                "margin": 8,
              },
            },
            "ref": null,
            "rendered": "LOGIN THROUGH FACEBOOK",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={1}
            style={
              Object {
                "marginBottom": 100,
              }
            }
          />,
          <TouchableOpacity
            activeOpacity={0.2}
            id="login-button"
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3b5999",
                "borderRadius": 4,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            &gt;
            <Image
              source={1}
              style={
                Object {
                  "margin": 8,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "margin": 8,
                }
              }
            >
              LOGIN THROUGH FACEBOOK
            </Text>
          </TouchableOpacity>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "marginBottom": 100,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              ">",
              <Image
                source={1}
                style={
                  Object {
                    "margin": 8,
                  }
                }
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "margin": 8,
                  }
                }
              >
                LOGIN THROUGH FACEBOOK
              </Text>,
            ],
            "id": "login-button",
            "onPress": [Function],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "#3b5999",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            ">",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
                "style": Object {
                  "margin": 8,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "LOGIN THROUGH FACEBOOK",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "margin": 8,
                },
              },
              "ref": null,
              "rendered": "LOGIN THROUGH FACEBOOK",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
